### PR TITLE
Combobox dialog example fix

### LINKED ISSUE #1942: Combobox Dialog Checkbox on doc site are broken
Checkboxes in Storybook work fine.

http://design-system-react-components.herokuapp.com/?selectedKind=SLDSCombobox&selectedStory=Dialog&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-pane

However, checkboxes in the Combobox dialog on the doc site do not work correctly: https://react.lightningdesignsystem.com/components/comboboxes/ (it's the last example)

This appears to be a checkbox event issue. The event callback at https://github.com/salesforce/design-system-react/blob/master/components/combobox/__examples__/dialog.jsx#L38 is referencing the checkbox component ref/target which is a React component, so target does not work from my understanding. Please see console for debugging.

```
Uncaught TypeError: Cannot read property 'checked' of undefined
    at Example.handleCheckboxChange (eval at executeCode (examples.jsx:172), <anonymous>:52:15)
    at onChange (eval at executeCode (examples.jsx:172), <anonymous>:126:20)
    at index.js:232
    at Object.i (react-dom.production.min.js:14)
    at o (react-dom.production.min.js:15)
    at a (react-dom.production.min.js:15)
    at u (react-dom.production.min.js:17)
    at p (react-dom.production.min.js:18)
    at Array.forEach (<anonymous>)
    at d (react-dom.production.min.js:18)
```

### PR DESCRIPTION
**Pull request by leandrobrunner**
Fixes #1942

### Additional description

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [x] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [x] `npm run lint:fix` has been run and linting passes.
* [x] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [x] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [x] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [x] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [x] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

**Comment by leandrobrunner** [comment]
Related conversation #1998

File: components/combobox/__examples__/dialog.jsx [hunk 1]
```diff
@@ -35,9 +35,9 @@ class Example extends React.Component {
 		return inputValue;
 	};
 
-	handleCheckboxChange(target, value) {
-		const { checked } = this.state;
-		if (target.checked) {
+	handleCheckboxChange(targetChecked, target, value) {
+		const checked = this.state.checked;
```
Comment (pulkonet) [inline components/combobox/__examples__/dialog.jsx]: Please use the object destructuring syntax here. This will also pass the lint issue in travis CI

**Comment by interactivellama** [comment]
The new checkbox `/components/checkbox` should be used. I think the API is using the old Checkbox API though.  The latest API for `onChange` is as follows. https://github.com/salesforce/design-system-react/blob/master/components/checkbox/index.jsx#L112

Please do not import the old checkbox. That code will then be copied into consumers' applications. The current examples looks like it is using the old API, but importing the newer Checkbox

```
	/**
	 * This event fires when the Checkbox changes. Passes in `event, { checked }`. This used to be `checked, event, { checked }`.
	 */
```

**Comment by leandrobrunner** [comment]
`@salesforce/design-system-react/components/checkbox` is pointing to `forms/checkbox`, the use of the new Checkbox API breaks the website example. https://github.com/salesforce/design-system-react/blob/24825aa68e981f6e7fbbef95f51f3c50e0a1ccc5/components/index.js#L84

**Comment by leandrobrunner** [comment]
The web https://react.lightningdesignsystem.com is importing from index, that point to `forms/checkbox` and the examples that use the new API dont work.

https://github.com/salesforce/design-system-react/blob/24825aa68e981f6e7fbbef95f51f3c50e0a1ccc5/components/index.js#L84

Any public repository to check it?

**Comment by interactivellama** [comment]
You are correct. As you proposed, we need to use the old API of the Checkbox until the default export works.

**Comment by interactivellama** [comment]
No, the site is not public at this time. I'll check it locally though on my machine this week.

**Comment by interactivellama** [comment]
It's working for me locally.

Screen Shot 2019-06-10 at 4 48 20 PM

**Pull request merged**